packages/devtools-connect/src/fast-failure-connect.ts

```diff
@@ -0,0 +1,13 @@
+// It probably makes sense to put this into its own package/repository once
+// other tools start using it.
+
+export function isFastFailureConnectionError(error: Error): boolean {
+  switch (error.name) {
+    case 'MongoNetworkError':
+      return /\b(ECONNREFUSED|ENOTFOUND|ENETUNREACH)\b/.test(error.message);
```

Comment:
```suggestion
      return /\b(ECONNREFUSED|ENOTFOUND|ENETUNREACH|EINVAL)\b/.test(error.message);
```

might as well address https://jira.mongodb.org/browse/MONGOSH-1717 while we're already in the process of making changes 😄 

(but feel free to ignore this if it's easier 🙂)

Reply:
sure! I was able to reproduce this with https://github.com/mongodb-js/mongosh/pull/1832, we can add this test once devtools-connect is updated